.github/workflows/include/rust-wasm-setup/action.yml

```diff
@@ -0,0 +1,16 @@
+name: Rust + WASM common deps
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set default toolchain
+      shell: bash
+      run: rustup default stable
+
+    - name: Install wasm-bindgen, wasmp-opt and wasm-pack
+      shell: bash
+      run: cargo install wasm-bindgen-cli wasm-opt wasm-pack
```

Comment:
It takes about 7 minutes to compile these on CI while previously they were installed as pre-compiled binaries. Can we solve this somehow? Maybe even `nix profile install` them (kinda contradicts the title of this PR but arguably not against the spirit of it because using nix as a classic imperative package manager doesn't really differ from any other package manager).

Reply:
We should check if all of these publish binaries as GitHub releases, then we could use `cargo binstall` instead of `cargo install`.

Reply:
https://github.com/cargo-bins/cargo-binstall

Reply:
What if instead we use a cache of .cargo/bin ?

Reply:
Cargo-binstall can also work

Reply:
Holy shit, Just saw you suggested the same

Reply:
That could work. What would we update it?

Reply:
Moving to `cargo binstall`

Reply:
Could we also bre-build a docker image with all dependencies installed?

Reply:
We don't need `wasm-pack` anymore, btw

Reply:
> Holy shit, Just saw you suggested the same

most of your comments didn't load for me too fwiw, I only saw the one about caching `.cargo/bin`